memory.c

```diff
@@ -27,13 +27,22 @@
 
 void memory_protect(void)
 {
+	// Reference STM32F205 Flash programming manual revision 5 http://www.st.com/resource/en/programming_manual/cd00233952.pdf
+	// Section 2.6 Option bytes
 	//                     set RDP level 2                   WRP for sectors 0 and 1
 	if ((((*OPTION_BYTES_1) & 0xFFFF) == 0xCCFF) && (((*OPTION_BYTES_2) & 0xFFFF) == 0xFFFC)) {
```

Comment:
Hmm, on second look, you might be right about bits 1:0 of the option bytes at 0x1FFF C000. The table I mentioned shows them as 0x1. So maybe that means OPTSTRT should be 0. That would make sense.